pkg/host/internal/sriov/sriov.go

```diff
@@ -467,76 +502,184 @@ func (s *sriov) ConfigSriovInterfaces(storeManager store.ManagerInterface,
+	}
+	log.Log.V(2).Info("ConfigSriovInterfacesInParallel(): sriov configuration finished")
+	return nil
+}
+
+func (s *sriov) resetSriovInterfacesInParallel(storeManager store.ManagerInterface, interfaces []sriovnetworkv1.InterfaceExt) error {
+	var result error
+	errChannel := make(chan error)
+	interfacesToReset := 0
+	for ifaceIndex := range interfaces {
+		interfacesToReset += 1
+		go func(iface *sriovnetworkv1.InterfaceExt) {
+			if err := s.checkForConfigAndReset(*iface, storeManager); err != nil {
+				log.Log.Error(err, "resetSriovInterfacesInParallel(): fail to configure sriov interface. resetting interface.", "address", iface.PciAddress)
+				errChannel <- err
+			} else {
+				errChannel <- nil
 			}
+		}(&interfaces[ifaceIndex])
+	}
 
-			if pfStatus.ExternallyManaged {
-				log.Log.Info("SyncNodeState(): PF name with pci address was externally created skipping the device reset",
-					"pf-name", ifaceStatus.Name,
-					"address", ifaceStatus.PciAddress)
-				continue
+	for i := 0; i < interfacesToReset; i++ {
+		errMsg := <-errChannel
+		result = errors.Join(result, errMsg)
+	}
+	if result != nil {
+		log.Log.Error(result, "resetSriovInterfacesInParallel(): fail to reset sriov interface")
+		return result
+	}
+	log.Log.V(2).Info("resetSriovInterfacesInParallel(): sriov reset finished")
+
+	return nil
+}
+
+func (s *sriov) configSriovInterfaces(storeManager store.ManagerInterface, interfaces []interfaceToConfigure) error {
+	for _, iface := range interfaces {
+		if err := s.ConfigSriovDevice(&iface.iface, &iface.ifaceStatus); err != nil {
+			log.Log.Error(err, "ConfigSriovInterfaces(): fail to configure sriov interface. resetting interface.", "address", iface.iface.PciAddress)
+			if iface.iface.ExternallyManaged {
+				log.Log.Info("ConfigSriovInterfaces(): skipping device reset as the nic is marked as externally created")
 			} else {
-				err = s.udevHelper.RemoveUdevRule(ifaceStatus.PciAddress)
-				if err != nil {
-					return err
+				if resetErr := s.ResetSriovDevice(iface.ifaceStatus); resetErr != nil {
+					log.Log.Error(resetErr, "ConfigSriovInterfaces(): failed to reset on error SR-IOV interface")
 				}
 			}
+			return err
+		}
 
-			if err = s.ResetSriovDevice(ifaceStatus); err != nil {
-				return err
-			}
+		// Save the PF status to the host
+		err := storeManager.SaveLastPfAppliedStatus(&iface.iface)
+		if err != nil {
+			log.Log.Error(err, "ConfigSriovInterfaces(): failed to save PF applied config to host")
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *sriov) resetSriovInterfaces(storeManager store.ManagerInterface, interfaces []sriovnetworkv1.InterfaceExt) error {
+	for _, iface := range interfaces {
+		if err := s.checkForConfigAndReset(iface, storeManager); err != nil {
+			log.Log.Error(err, "resetSriovInterfaces(): fail to configure sriov interface. resetting interface.", "address", iface.PciAddress)
+			return err
+		}
+	}
+	log.Log.V(2).Info("resetSriovInterfaces(): sriov reset finished")
```

Comment:
can you add additional logs when starting and finishing configuring / resetting an interface both in parallel and serially ?

Reply:
done